src/main.rs

```diff
@@ -481,8 +543,19 @@ impl HabitCtl {
             .collect();
         done.retain(|value| *value);
 
+        let mut skip: Vec<bool> = self
+            .habits
+            .iter()
+            .map(|habit| {
+                let status = self.day_status(&habit, &score_date);
+                habit.every_days > 0
+                    && (status == DayStatus::Skipped || status == DayStatus::Skipified)
+            })
+            .collect();
+        skip.retain(|value| *value);
```

Comment:
This is obviously a dark magic incantation, but I'm supposing it's some sort of filtering according to the booleans in the `skip` vector above?

Reply:
Retain basically says what you can keep in the vector yes, but I have ti admit I am a little confused by this line myself since it seems to say to keep the pointers of the values in the vector. It is a diredt copy from the working function above for Done. But yes, voodoo indeed. From my reading p n this in Rust, still assuming we want it so we're dealing with pointers rather than the vectors itself (which *does* make sense, considering...).